extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/sql/resources/SqlStatementResource.java

```diff
@@ -879,6 +881,16 @@ private <T> T contactOverlord(final ListenableFuture<T> future)
       return FutureUtils.getUnchecked(future, true);
     }
     catch (RuntimeException e) {
+      if (e.getCause() instanceof HttpResponseException) {
+        HttpResponseException httpResponseException = (HttpResponseException) e.getCause();
+        if (httpResponseException.getResponse() != null && httpResponseException.getResponse().getResponse().getStatus()
+                                                                                .equals(HttpResponseStatus.NOT_FOUND)) {
+          // since we get a 404, we mark the request as an invalid input. This code path is generally triggered when user passes a `queryId` which is not found on the overlord.
+          throw DruidException.forPersona(DruidException.Persona.DEVELOPER)
+                              .ofCategory(DruidException.Category.INVALID_INPUT)
+                              .build("Unable to contact overlord " + e.getMessage());
```

Comment:
```suggestion
      throw InvalidInput.exception(
          e, "Unable to contact the overlord for the query id [%s]
      );
```

Reply:
Also, apart from the above, one more suggestion - I think since we are rethrowing the error, we should not chomp down `e` (or `e.getCause`), and attach them as the cause of the new exception.

Reply:
If it is an invalid input, why is the error message "unable to contact overlord"?

Reply:
I'm not very familiar with the DruidException, but this is thrown when the overlord is not reachable. Is there a reason that `InvalidInput` is the best fit here?